specs/agents/configuration.md

```diff
@@ -10,6 +10,51 @@ Here's a list of the config options across all agents, their types, default valu
 
 They are provided as environment variables but depending on the language there might be several feasible ways to let the user tweak them. For example besides the environment variable `ELASTIC_APM_SERVER_URL`, the Node.js Agent might also allow the user to configure the server URL via a config option named `serverUrl`, while the Python Agent might also allow the user to configure it via a config option named `server_url`.
 
+### Configuration Source Precedence
+
+Configuration can be provided via a number of sources. The sources and their
```

Comment:
How strict is this part of the doc? The reason I ask is that the .NET Agent integrates with a standard config system from Microsoft (some docs and code [here](https://docs.microsoft.com/en-us/dotnet/core/extensions/configuration-providers#json-configuration-provider)) and with that the precedence is defined by the user (it's the order in which config providers are registered), and it's very common that e.g. config files have higher precedence than environment variables.  Having said that, central config has the higher precedence in the .NET Agent, so it can override any other config source.

So, if things like that is fine, I'd suggest working with `MAY` and `MUST`. Would it maybe ok to say something like: "Central config MUST have the highest precedence and Default values MUST have the lowest precedence. Agents MAY follow the following precedence for other config sources: [put the remaining sources in current order here]". I think forcing this precedence is reasonable and makes sense, but forcing the currently suggested precedence from this PR would be (at least in .NET) harder with not much benefit.

Reply:
I used `SHOULD` instead of `MAY`, but otherwise followed your advice. I think `SHOULD` is the better fit here, because as far as I know, .NET is alone in not following this pattern.

Reply:
The Java agent is not following this order as well 🙂 .
Like central configuration, in the Java agent configs are made dynamic through config files, which means that it must have higher precedence than env or code. In general, our approach was that the "further" the config vehicle is from the code, the higher precedence it would get, but not necessarily this makes sense to all.

I would also prefer `MAY`, or even leave the in-between undefined, but we can also use `SHOULD` and not follow it in the Java agent as well.

Reply:
I'll use `MAY` since we now have two examples. :)